### PR TITLE
Module 3.home work

### PR DESCRIPTION
**Pull request by taraswww777**

### CODE REVIEW

File: src/app/app.module.ts [hunk 1]
```diff
@@ -46,6 +46,9 @@ import { LoaderComponent } from './loader/loader.component';
 		FormsModule,
 		ReactiveFormsModule
 	],
+	schemas: [
```
Comment (gairon) [inline src/app/app.module.ts]: Why do you need the schema?

File: src/app/breadcrumbs/breadcrumbs.component.spec.ts [hunk 2]
```diff
@@ -1,25 +1,31 @@
...
+			.compileComponents();
+	}));
 
-  beforeEach(() => {
-    fixture = TestBed.createComponent(BreadcrumbsComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
-  });
+	beforeEach(() => {
+		fixture = TestBed.createComponent(BreadcrumbsComponent);
```
Comment (gairon) [inline src/app/breadcrumbs/breadcrumbs.component.spec.ts]: you can combine this beforeEach and previous into one

File: src/app/courses-list/courses-list.component.spec.ts [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
+	let component: CoursesListComponent;
+	let fixture: ComponentFixture<CoursesListComponent>;
 
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      declarations: [ CoursesListComponent ]
-    })
-    .compileComponents();
-  }));
+	beforeEach(async(() => {
```
Comment (gairon) [inline src/app/courses-list/courses-list.component.spec.ts]: `async` isn't required here. There is no async operations.

File: src/app/courses-list/courses-list.component.spec.ts [hunk 4]
```diff
@@ -1,25 +1,27 @@
...
+import {async, ComponentFixture, TestBed} from '@angular/core/testing';
 
-import { CoursesListComponent } from './courses-list.component';
+import {CoursesListComponent} from './courses-list.component';
+import {CUSTOM_ELEMENTS_SCHEMA} from "@angular/core";
 
 describe('CoursesListComponent', () => {
-  let component: CoursesListComponent;
-  let fixture: ComponentFixture<CoursesListComponent>;
+	let component: CoursesListComponent;
```
Comment (gairon) [inline src/app/courses-list/courses-list.component.spec.ts]: You can test 
1) that every cours are present in the list.
2) a user can remove a course and the count of items will decrease

File: src/app/courses-load-more/courses-load-more.component.ts [hunk 5]
```diff
@@ -18,9 +18,7 @@ export class CoursesLoadMoreComponent implements OnInit {
 	}
 
 	onLoadMore() {
-		console.log('onLoadMore:');
 		this.showLoader = true;
-		setTimeout(() => {
```
Comment (gairon) [inline src/app/courses-load-more/courses-load-more.component.ts]: Try to implement async test using ` tick()`

File: src/app/page-courses/page-courses.component.spec.ts [hunk 6]
```diff
@@ -19,6 +37,10 @@ describe('PageCoursesComponent', () => {
 		fixture.detectChanges();
 	});
 
+	it('listCourses', () => {
+		expect(component.listCourses).toEqual(arr)
```
Comment (gairon) [inline src/app/page-courses/page-courses.component.spec.ts]: This test needs to be rewriten after implementing a service for cources receiving

File: src/app/app.module.ts (hunk 1 above)
Comment (taraswww777) [reply src/app/app.module.ts]: I thought in this place need to use schemas

File: src/app/page-courses/page-courses.component.spec.ts (hunk 6 above)
Comment (taraswww777) [reply src/app/page-courses/page-courses.component.spec.ts]: Of course

**Pull request merged**